recipes/pthreadpool/all/conanfile.py

```diff
@@ -36,35 +36,24 @@ def config_options(self):
 
     def configure(self):
         if self.options.shared:
-            try:
-                del self.options.fPIC
-            except Exception:
-                pass
-        try:
-            del self.settings.compiler.libcxx
-        except Exception:
-            pass
-        try:
-            del self.settings.compiler.cppstd
-        except Exception:
-            pass
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
 
     def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("fxdiv/cci.20200417")
+        self.requires("fxdiv/cci.20200417", transitive_headers=True)
```

Comment:
Is this the case?
The single public header from `pthreadpool`, does not have a transitive header dependency on `fxdiv`, I can only see references to:
```
#include <stddef.h>
#include <stdint.h>
```

On the other hand, the library authors explicitly mark it as a _private_ dependency:
https://github.com/Maratyszcza/pthreadpool/blob/master/CMakeLists.txt#L164

So I'm unsure as to what motivates this change.

Reply:
Yes, you're right: https://github.com/Maratyszcza/pthreadpool/blob/b4589998be9a0f794236cf46f1b5b232b2b15ca3/include/pthreadpool.h (this is the commit packaged here)